packages/web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx

```diff
@@ -0,0 +1,515 @@
+      this.currentPage = page;
+      this.lowerBound = (page - 1) * Number(this.itemsPerPage) + 1;
+      this.setUpperBound();
+    }
+  };
+
+  private goToPage = () => {
+    const input = this.el.shadowRoot.querySelector(
+      this.TEXT_FIELD
+    ) as HTMLIcTextFieldElement;
+    const page = Number(input.value);
+    const tooltip = this.el.shadowRoot.querySelector("ic-tooltip");
+    if (page <= this.totalPages && page > 0) {
+      this.changePage(page);
+      const pagination = this.el.shadowRoot.querySelector(
+        this.PAGINATION
+      ) as HTMLIcPaginationElement;
+      pagination.setCurrentPage(page);
+      this.currentPage = page;
+      input.value = "";
+      this.icPageChange.emit({ value: page });
+      tooltip.displayTooltip(false, false);
+      input.validationStatus = "";
+    } else {
+      this.inputError = this.INVALID_PAGE_ERROR;
+      input.validationStatus = "error";
+      input.setFocus();
+    }
+  };
+
+  private handleBlur = () => {
+    const textField = this.el.shadowRoot?.querySelector(
+      this.TEXT_FIELD
+    ) as HTMLIcTextFieldElement;
+    const tooltip = this.el.shadowRoot?.querySelector(
+      this.TOOLTIP
+    ) as HTMLIcTooltipElement;
+    if (
+      (Number(textField.value) <= this.totalPages &&
+        Number(textField.value) > 0) ||
+      textField.value === ""
+    ) {
+      textField.validationStatus = "";
+    }
+    tooltip.displayTooltip(false, false);
+  };
+
+  private handleFocus = () => {
+    const textField = this.el.shadowRoot?.querySelector(
+      this.TEXT_FIELD
+    ) as HTMLIcTextFieldElement;
+    const tooltip = this.el.shadowRoot?.querySelector(
+      this.TOOLTIP
+    ) as HTMLIcTooltipElement;
+    if (textField.validationStatus === "error") {
+      tooltip.displayTooltip(true, true);
+    }
+  };
+
+  private handleInputChange = () => {
+    const tooltip = this.el.shadowRoot.querySelector(
+      this.TOOLTIP
+    ) as HTMLIcTooltipElement;
+    const textField = this.el.shadowRoot.querySelector(
+      this.TEXT_FIELD
+    ) as HTMLIcTextFieldElement;
+    const inputValue = parseInt(textField.value);
+
+    if (inputValue > this.totalPages || inputValue <= 0) {
+      this.inputError = this.INVALID_PAGE_ERROR;
+      tooltip.displayTooltip(true, true);
+      textField.validationStatus = "error";
+      textField.focus();
+    }
+  };
+
+  private handleKeydown = (ev: KeyboardEvent) => {
+    const tooltip = this.el.shadowRoot.querySelector(
+      this.TOOLTIP
+    ) as HTMLIcTooltipElement;
+    const textField = this.el.shadowRoot.querySelector(
+      this.TEXT_FIELD
+    ) as HTMLIcTextFieldElement;
+
+    if (ev.key === "Enter") {
+      if (textField.validationStatus === "error") {
+        tooltip.displayTooltip(false, false);
+        setTimeout(() => tooltip.displayTooltip(true, true), 50);
+      } else {
+        this.goToPage();
+      }
+    } else if (ev.key === "Backspace") {
+      tooltip.displayTooltip(false, true);
```

Comment:
Does this need to persist if you are amending the value while it is not longer on show? I might not be testing this correctly.